camera: Initialization fixes (cosmetic)

Quallenauge opened the pull request:
### Part1: camera: msm8976-camera: Add camss_top_ahb_clk_src to CPP node

This makes the clock sequence to be correct at boot and will
stop the kernel warning at CPP first init.

Inspired by
https://github.com/sonyxperiadev/kernel/commit/ce879415318cd4155c1f18ddfade364414341f65

Clock rates where taken from the clock definitions:
https://source.codeaurora.org/quic/chrome/kernel/msm-3.10/tree/drivers/clk/qcom/clock-gcc-8976.c?h=LA.BR.1.3.7-05010-8976.0#n891

Fixes warning when initializing camera interface:
```
[    1.114223] msm_cci_get_clk_info: clock-names[0][0] = camss_top_ahb_clk
[    1.114345] msm_cci_get_clk_info: clk_rate[0][0] = -1
[    1.114410] msm_cci_get_clk_info: clock-names[0][1] = cci_src_clk
[    1.114531] msm_cci_get_clk_info: clk_rate[0][1] = 19200000
[    1.114597] msm_cci_get_clk_info: clock-names[0][2] = cci_ahb_clk
[    1.114716] msm_cci_get_clk_info: clk_rate[0][2] = -1
[    1.114781] msm_cci_get_clk_info: clock-names[0][3] = camss_cci_clk
[    1.114901] msm_cci_get_clk_info: clk_rate[0][3] = -1
[    1.114965] msm_cci_get_clk_info: clock-names[0][4] = camss_ahb_clk
[    1.115083] msm_cci_get_clk_info: clk_rate[0][4] = -1
[    1.115148] msm_cci_get_clk_info: clock-names[0][5] = ispif_ahb_clk
[    1.115268] msm_cci_get_clk_info: clk_rate[0][5] = 61540000
[    1.115332] msm_cci_get_clk_info: clock-names[1][0] = camss_top_ahb_clk
[    1.115452] msm_cci_get_clk_info: clk_rate[1][0] = -1
[    1.115516] msm_cci_get_clk_info: clock-names[1][1] = cci_src_clk
[    1.115636] msm_cci_get_clk_info: clk_rate[1][1] = 37500000
[    1.115702] msm_cci_get_clk_info: clock-names[1][2] = cci_ahb_clk
[    1.115822] msm_cci_get_clk_info: clk_rate[1][2] = -1
[    1.115887] msm_cci_get_clk_info: clock-names[1][3] = camss_cci_clk
[    1.116007] msm_cci_get_clk_info: clk_rate[1][3] = -1
[    1.116072] msm_cci_get_clk_info: clock-names[1][4] = camss_ahb_clk
[    1.116192] msm_cci_get_clk_info: clk_rate[1][4] = -1
[    1.116257] msm_cci_get_clk_info: clock-names[1][5] = ispif_ahb_clk
[    1.116377] msm_cci_get_clk_info: clk_rate[1][5] = 61540000
[    1.118703] adp1660 i2c_add_driver success
[    1.119282] msm_torch_create_classdev:135 Invalid fctrl->torch_trigger[0]
[    1.119414] qcom,camera-flash: probe of qcom,camera-flash.105 failed with error -22
[    1.119620] msm_flash_lm3642_init entry
[    1.121655] msm_cci_init: Failed in getting TOP gdscr regulator handle
[    1.121802] ------------[ cut here ]------------
[    1.121940] WARNING: at /../drivers/clk/qcom/clock-local2.c:216 rcg_clk_enable+0x50/0xa0()
[    1.122062] Attempting to prepare camss_top_ahb_clk_src before setting its rate. Set the rate first!
[    1.122183] Modules linked in:
[    1.122312] CPU: 4 PID: 1 Comm: swapper/0 Tainted: G        W    3.10.108_LA.BR.1.3.7_rb1.11+ #45
[    1.122433] Call trace:
[    1.122500] [<ffffffc000087dcc>] dump_backtrace+0x0/0x26c
[    1.122621] [<ffffffc00008804c>] show_stack+0x14/0x1c
[    1.122688] [<ffffffc000d0da50>] dump_stack+0x20/0x28
[    1.122810] [<ffffffc00009dda0>] warn_slowpath_common+0x78/0x9c
[    1.122875] [<ffffffc00009de14>] warn_slowpath_fmt+0x50/0x58
[    1.122996] [<ffffffc0009fc900>] rcg_clk_enable+0x50/0xa0
[    1.123061] [<ffffffc0009f9afc>] clk_enable+0x174/0x1b4
[    1.123180] [<ffffffc0009f9a24>] clk_enable+0x9c/0x1b4
[    1.123247] [<ffffffc000672084>] msm_cam_clk_enable+0x164/0x290
[    1.123367] [<ffffffc000670be8>] msm_cci_config+0x530/0x12e0
[    1.123432] [<ffffffc000671a40>] msm_cci_subdev_ioctl+0xa8/0xf0
[    1.123554] [<ffffffc000673e54>] msm_sensor_cci_i2c_util+0x78/0xc8
[    1.123620] [<ffffffc000678df0>] msm_camera_power_up+0x4e4/0x764
[    1.123740] [<ffffffc000686ed0>] msm_eeprom_platform_probe+0x3f8/0x5f8
[    1.123807] [<ffffffc000489420>] platform_drv_probe+0x18/0x20
[    1.123928] [<ffffffc000487f90>] driver_probe_device+0x164/0x374
[    1.123994] [<ffffffc000488250>] __driver_attach+0x64/0x90
[    1.124115] [<ffffffc000486d80>] bus_for_each_dev+0x74/0x90
[    1.124181] [<ffffffc0004878d0>] driver_attach+0x20/0x28
[    1.124302] [<ffffffc000487480>] bus_add_driver+0x120/0x240
[    1.124368] [<ffffffc000488870>] driver_register+0x94/0x110
[    1.124491] [<ffffffc0004898f8>] platform_driver_register+0x5c/0x64
[    1.124557] [<ffffffc00048993c>] platform_driver_probe+0x24/0x9c
[    1.124680] [<ffffffc0014555dc>] msm_eeprom_init_module+0x50/0xd8
[    1.124746] [<ffffffc000081910>] do_one_initcall+0xb4/0x158
[    1.124806] synaptics_rmi4_sensor_report: spontaneous reset detected
[    1.124982] [<ffffffc00141e908>] kernel_init_freeable+0x148/0x1e8
[    1.125038] [<ffffffc000d038ec>] kernel_init+0x14/0xcc
[    1.125159] ---[ end trace da227214a82491b8 ]---
```

### Part2: camera: Add stub regulators for camera's gdscr.

Although regulators are optional in camera drivers,
there is a warning each time the camera initializes,
which indicates an error, by not getting the TOP gdscr regulator handle.

[    1.118752] msm_cci_init: Failed in getting TOP gdscr regulator handle

MSM8976 has no additional gdscr (gdsc) vregs,
so stub them out and silence the error.

Inspired by:
https://github.com/sonyxperiadev/kernel/commit/59813018215991829406ad85c988ca2aa33ece5c

kholk committed:
camera: msm8976-camera: Add camss_top_ahb_clk_src to CPP node

This makes the clock sequence to be correct at boot and will
stop the kernel warning at CPP first init.

Inspired by
https://github.com/sonyxperiadev/kernel/commit/ce879415318cd4155c1f18ddfade364414341f65

Clock rates where taken from the clock definitions:
https://source.codeaurora.org/quic/chrome/kernel/msm-3.10/tree/drivers/clk/qcom/clock-gcc-8976.c?h=LA.BR.1.3.7-05010-8976.0#n891

Fixes warning when initializing camera interface:
[    1.121802] ------------[ cut here ]------------
[    1.121940] WARNING: at ../drivers/clk/qcom/clock-local2.c:216 rcg_clk_enable+0x50/0xa0()
[    1.122062] Attempting to prepare camss_top_ahb_clk_src before setting its rate. Set the rate first!
[    1.122183] Modules linked in:
[    1.122312] CPU: 4 PID: 1 Comm: swapper/0 Tainted: G        W    3.10.108_LA.BR.1.3.7_rb1.11+ #45
[    1.122433] Call trace:
[    1.122500] [<ffffffc000087dcc>] dump_backtrace+0x0/0x26c
[    1.122621] [<ffffffc00008804c>] show_stack+0x14/0x1c
[    1.122688] [<ffffffc000d0da50>] dump_stack+0x20/0x28
[    1.122810] [<ffffffc00009dda0>] warn_slowpath_common+0x78/0x9c
[    1.122875] [<ffffffc00009de14>] warn_slowpath_fmt+0x50/0x58
[    1.122996] [<ffffffc0009fc900>] rcg_clk_enable+0x50/0xa0
[    1.123061] [<ffffffc0009f9afc>] clk_enable+0x174/0x1b4
[    1.123180] [<ffffffc0009f9a24>] clk_enable+0x9c/0x1b4
[    1.123247] [<ffffffc000672084>] msm_cam_clk_enable+0x164/0x290
[    1.123367] [<ffffffc000670be8>] msm_cci_config+0x530/0x12e0
[    1.123432] [<ffffffc000671a40>] msm_cci_subdev_ioctl+0xa8/0xf0
[    1.123554] [<ffffffc000673e54>] msm_sensor_cci_i2c_util+0x78/0xc8
[    1.123620] [<ffffffc000678df0>] msm_camera_power_up+0x4e4/0x764
[    1.123740] [<ffffffc000686ed0>] msm_eeprom_platform_probe+0x3f8/0x5f8
[    1.123807] [<ffffffc000489420>] platform_drv_probe+0x18/0x20
[    1.123928] [<ffffffc000487f90>] driver_probe_device+0x164/0x374
[    1.123994] [<ffffffc000488250>] __driver_attach+0x64/0x90
[    1.124115] [<ffffffc000486d80>] bus_for_each_dev+0x74/0x90
[    1.124181] [<ffffffc0004878d0>] driver_attach+0x20/0x28
[    1.124302] [<ffffffc000487480>] bus_add_driver+0x120/0x240
[    1.124368] [<ffffffc000488870>] driver_register+0x94/0x110
[    1.124491] [<ffffffc0004898f8>] platform_driver_register+0x5c/0x64
[    1.124557] [<ffffffc00048993c>] platform_driver_probe+0x24/0x9c
[    1.124680] [<ffffffc0014555dc>] msm_eeprom_init_module+0x50/0xd8
[    1.124746] [<ffffffc000081910>] do_one_initcall+0xb4/0x158
[    1.124806] synaptics_rmi4_sensor_report: spontaneous reset detected
[    1.124982] [<ffffffc00141e908>] kernel_init_freeable+0x148/0x1e8
[    1.125038] [<ffffffc000d038ec>] kernel_init+0x14/0xcc
[    1.125159] ---[ end trace da227214a82491b8 ]---

Co-authored-by: Quallenauge <Hamsi2k@freenet.de>

Gabriel85 commented:
OK, good! What about the other similar error? :)

Quallenauge commented:
Without an specific error I'm not able to know what for an error is meant.
This PR fixes the above warning.

Gabriel85 commented:
> [89475.664017]  msm_csid_init: Failed in getting TOP gdscr regulator handle
> [89475.665034] ------------[ cut here ]------------
> [89475.665131] WARNING: at drivers/clk/qcom/clock-local2.c:216 rcg_clk_enable+0x50/0xa0()
> [89475.665221] Attempting to prepare csi2_clk_src before setting its rate. Set the rate first!
> [89475.665227] CPU: 5 PID: 28120 Comm: CAM_sensor Tainted: G        W    3.10.108_LA.BR.1.3.7_rb1.11-gb5400113245-dirty #26
> [89475.665231] Call trace:
> [89475.665239] [<ffffffc000087ecc>] dump_backtrace+0x0/0x12c
> [89475.665244] [<ffffffc00008800c>] show_stack+0x14/0x1c
> [89475.665249] [<ffffffc000ce98a8>] dump_stack+0x20/0x28
> [89475.665254] [<ffffffc00009d2c4>] warn_slowpath_common+0x74/0xa0
> [89475.665257] [<ffffffc00009d36c>] warn_slowpath_fmt+0x50/0x58
> [89475.665260] [<ffffffc0009ecd30>] rcg_clk_enable+0x50/0xa0
> [89475.665265] [<ffffffc0009e9e64>] clk_enable+0x14c/0x1b4
> [89475.665268] [<ffffffc0009e9db4>] clk_enable+0x9c/0x1b4
> [89475.665273] [<ffffffc000661134>] msm_cam_clk_enable+0x170/0x298
> [89475.665277] [<ffffffc000668c00>] msm_csiphy_subdev_ioctl+0x398/0xb40
> [89475.665281] [<ffffffc000669404>] msm_csiphy_subdev_do_ioctl+0x5c/0x68
> [89475.665287] [<ffffffc000600b10>] video_usercopy+0x260/0x400
> [89475.665291] [<ffffffc000668484>] msm_csiphy_subdev_fops_ioctl+0x14/0x1c
> [89475.665295] [<ffffffc00060a084>] v4l2_compat_ioctl32+0x530/0x548
> [89475.665300] [<ffffffc0001bb9b0>] compat_sys_ioctl+0x10c/0x1588
> [89475.665303] ---[ end trace da227214a82491c1 ]---

Quallenauge commented:
If you mean
>  msm_csid_init: Failed in getting TOP gdscr regulator handle

no. It's not part of the PR.

Gabriel85 commented:
Yes I know that. ;)
But it is the same kind of error, with the same kind of solution..

Quallenauge committed:
camera: Add clk_csi2_clk_src to CPP node

This makes the clock sequence to be correct at boot and will
stop the kernel warning at CPP first init.

The clock rate is set to 200000000,
which is within the defined frequency table:
https://source.codeaurora.org/quic/chrome/kernel/msm-3.10/tree/drivers/clk/qcom/clock-gcc-8976.c?h=LA.BR.1.3.7-05010-8976.0#n710
and also used at other places:
(4th element of the following array)
https://source.codeaurora.org/quic/chrome/kernel/msm-3.10/tree/arch/arm/boot/dts/qcom/msm8976-camera.dtsi?h=LA.BR.1.3.7-05010-8976.0#n140

kholk committed:
camera: Add stub regulators for camera's gdscr.

Although regulators are optional in camera drivers,
there is a warning each time the camera initializes,
which indicates an error, by not getting the TOP gdscr regulator handle.

[    1.118752] msm_cci_init: Failed in getting TOP gdscr regulator handle

MSM8976 has no additional gdscr (gdsc) vregs,
so stub them out and silence the error.

Inspired by:
https://github.com/sonyxperiadev/kernel/commit/59813018215991829406ad85c988ca2aa33ece5c

Co-authored-by: Quallenauge <Hamsi2k@freenet.de>

Quallenauge commented:
Added a new commit, which silences the msm_csid_init: Failed in getting TOP gdscr regulator handle
by adding a stub gdscr handle.

Gabriel85 commented:
Does the last patch fix this, too:
`Attempting to prepare csi2_clk_src before setting its rate. Set the rate first!`?

Quallenauge commented:
No, I saw this warning just in the last minutes.

Gabriel85 commented:
Sorry, my bad. Actually, I meant this one yesterday..

Quallenauge commented:
Added an commit to handle the warning:
> Attempting to prepare csi2_clk_src before setting its rate. Set the rate first!


https://github.com/Lenovo-YTX703-Devel/android_kernel_lenovo_msm8976/pull/13/commits/8dd60e0a054aa130ae13330684fd7364d6f1ba0a
